### ISSUE TITLE
This repository is cringe

### ISSUE THREAD

**Issue by orsondmc** [RandomGamingDev/EzCopenheimer#2]
This came to my attention and this is one of the cringerier cope clones I have ever seen.

**Comment by RandomGamingDev** [comment]
Lmao thanks ig. So ur the og Creator of copenheimer? Cool. Could u tell me how u came up with the idea and what your thoughts are on my reverse engineered version? Also, I'm not sure what u mean by "cringe", but this repo's designed more so for ease of use, educational purposes, and as a simple proof of concept rather than as an actual scanning software which is why it's not that fast, nor has a system for external nor automatic range handling or division to threads, although I do have plans to improve it in the future when I stop procrastinating. :D

**Comment by ghost** [comment]
> This came to my attention and this is one of the cringerier cope clones I have ever seen.

its come to my attention that i screwed ur mom last night

**Comment by JacobDevWasTaken** [comment]
> This came to my attention and this is one of the cringerier cope clones I have ever seen.

true

**Comment by ghost** [comment]
> Lmao thanks ig. So ur the og Creator of copenheimer? Cool. Could u tell me how u came up with the idea and what your thoughts are on my reverse engineered version? Also, I'm not sure what u mean by "cringe", but this repo's designed more so for ease of use, educational purposes, and as a simple proof of concept rather than as an actual scanning software which is why it's not that fast, nor has a system for external nor automatic range handling or division to threads, although I do have plans to improve it in the future when I stop procrastinating. :D

Your repository is literally integrated into Wurst, not yours specifically, but someone else's who is likely more intelligent than you to think this is even remotely close to Copenheimer.

**Comment by RandomGamingDev** [comment]
> > Lmao thanks ig. So ur the og Creator of copenheimer? Cool. Could u tell me how u came up with the idea and what your thoughts are on my reverse engineered version? Also, I'm not sure what u mean by "cringe", but this repo's designed more so for ease of use, educational purposes, and as a simple proof of concept rather than as an actual scanning software which is why it's not that fast, nor has a system for external nor automatic range handling or division to threads, although I do have plans to improve it in the future when I stop procrastinating. :D
> 
> You'r repo is quite litteraly integrated into wurst not yours specifically but someone else's who is likely more intelligent then you are to think this is even remotely close to Copenheimer

1. Grammar???
2. When did I ever say that this was made to be the same thing as Copenheimer? I explicitly state in the video and github that this is an educational tool (that's why it's designed in python in a way that's easily comprehensible instead of just using smth like masscan) and that I don't want people nuking servers.

**Comment by ghost** [comment]
> > > Lmao thanks ig. So ur the og Creator of copenheimer? Cool. Could u tell me how u came up with the idea and what your thoughts are on my reverse engineered version? Also, I'm not sure what u mean by "cringe", but this repo's designed more so for ease of use, educational purposes, and as a simple proof of concept rather than as an actual scanning software which is why it's not that fast, nor has a system for external nor automatic range handling or division to threads, although I do have plans to improve it in the future when I stop procrastinating. :D
> > 
> > 
> > You'r repo is quite litteraly integrated into wurst not yours specifically but someone else's who is likely more intelligent then you are to think this is even remotely close to Copenheimer
> 
> 1. Grammar???
> 2. When did I ever say that this was made to be the same thing as Copenheimer? I explicitly state in the video and github that this is an educational tool (that's why it's designed in python in a way that's easily comprehensible instead of just using smth like masscan) and that I don't want people nuking servers.

This 'educational' claim is crazy. Do you really believe anyone would use this for educational purposes? Why would you recreate something used for nuking servers and expect people not to misuse it? Are you out of your mind?

**Comment by ghost** [comment]
And I apologize if my grammar was not up to your standards. Criticizing '1 grammar' is absurd. Compared to me, you are insignificant. I didn't have time to correct myself because, unlike you, I have significant responsibilities in this world. 

^^

**Comment by FFUV** [comment]
Seriously, you think you’ve done something big with this repo? 😂 Even one of the main devs of Copenheimer is calling you out. 💀

**Comment by RandomGamingDev** [comment]
> > > > Lmao thanks ig. So ur the og Creator of copenheimer? Cool. Could u tell me how u came up with the idea and what your thoughts are on my reverse engineered version? Also, I'm not sure what u mean by "cringe", but this repo's designed more so for ease of use, educational purposes, and as a simple proof of concept rather than as an actual scanning software which is why it's not that fast, nor has a system for external nor automatic range handling or division to threads, although I do have plans to improve it in the future when I stop procrastinating. :D
> > > 
> > > 
> > > You'r repo is quite litteraly integrated into wurst not yours specifically but someone else's who is likely more intelligent then you are to think this is even remotely close to Copenheimer
> > 
> > 
> > 
> > 1. Grammar???
> > 2. When did I ever say that this was made to be the same thing as Copenheimer? I explicitly state in the video and github that this is an educational tool (that's why it's designed in python in a way that's easily comprehensible instead of just using smth like masscan) and that I don't want people nuking servers.
> 
> This 'educational' claim is crazy. Do you really believe anyone would use this for educational purposes? Why would you recreate something used for nuking servers and expect people not to misuse it? Are you out of your mind?

Creating or modelling a simplified example of something to teach about something is common practice. For instance, those teaching swordsmanship have wooden swords, those teaching how to make bullet proof materials teach a basic model for how guns work, and those teaching cybersecurity create a simplified overview, or actual piece of simplified malware. Yes, you can bash in someone's head with a wooden sword, yes you can create a zip gun to attack people with with an understanding of a simplified gun, and yes you can hack people with a piece of simplified malware. Is that the purpose, or intent of the tool though? No, and if you say otherwise we'll both know you're lying. Should these tools, then, just because of their danger be outlawed? No. It's just an example used for demonstration, which I've used for demonstrating how these things work not only through github and youtube, but with other people in real life as well.



> And I apologize if my grammar was not up to your standards. Criticizing '1 grammar' is absurd. Compared to me, you are insignificant. I didn't have time to correct myself because, unlike you, I have significant responsibilities in this world.
> 
> ^^

Let's be honest, the only reason I complained about your grammar was that it's near incomprehensible. I don't care about grammar much and make a ton of mistakes myself, but at least make what you say readable. Also, attacking me personally? Ad hominem.


> Seriously, you think you’ve done something big with this repo? 😂 Even one of the main devs of Copenheimer is calling you out. 💀

I never said I was doing anything big with the repo? So far multiple of your arguments have been the literal definition of a straw man fallacy. Also, the reason he called my repo cringe is likely the fact that this is an educational tool designed for simplicity of understanding instead of speed (this example is written in python with 3 common python libraries, isn't even asked to be run in something like pypy, and if I wanted something fast and damaging I could've just used massscan + filter which would've been multitudes of times easier), which is different from what his version and multiple other Copenheimer clones were designed for: aiming to be fast and to be able to cause damage.

**Comment by ghost** [comment]
This entire issue is cringe, you guys are arguing about a Minecraft tool. Someone close it and move on.

**Comment by RandomGamingDev** [comment]
> This entire issue is cringe, you guys are arguing about a Minecraft tool. Someone close it and move on.

Yeah, agreed. I wanted to leave it here as a joke, but it'll probably attract more people like @FFUV so I'll close it.

**Comment by FFUV** [comment]
> > This entire issue is cringe, you guys are arguing about a Minecraft tool. Someone close it and move on.
> 
> Yeah, agreed. I wanted to leave it here as a joke, but it'll probably attract more people like @FFUV so I'll close it.

Closing it sounds like you're running away from consequences.

Creating or modeling a simplified example of something is a common practice, but claiming that this project is purely "educational" is disingenuous. This project is a Minecraft hack to find and track people on servers, and it's a bad version at that. The potential for misuse is obvious, and to think people won't use this for malicious purposes is naive or intentionally ignorant.

And about the grammar—criticizing my grammar is absurd. Compared to me, you are insignificant. I didn't have time to correct myself because, unlike you, I have significant responsibilities in this world.

Seriously, you think you’ve done something big with this repo? 😂 Even one of the main devs of Copenheimer is calling you out. 💀

I've also noticed inconsistencies in what I've asked for and what you've done. Your claim that this is an educational tool designed for simplicity is questionable at best. If I wanted something fast and damaging, I could've just used masscan + filter, which would've been multitudes of times easier. This example is written in Python with 3 common Python libraries, isn't even asked to be run in something like PyPy, and it clearly doesn't aim for speed or efficiency, unlike other Copenheimer clones designed to cause damage.

By the way, I know you reported me, but it only shows your desperation to silence criticism instead of addressing the real issues.

Also, let's take a look at your code:

`from mcstatus import JavaServer
import threading
import asyncio

defaultPort = 25565
defaultTimeout = 1
waitForTimeout = 100
ipIters = [
               [
                    #Enter the IP ranges here like this: [[127, 0, 0, 1], [127, 0, 0, 1]],
		            #A thread will scan each
               ],
               [ # Since computer network buffers are limited, there are different ipIters which are looped over synchronously in order, thus this is the same as the previous one, it just runs afterwards
               ]
          ]
outputFileName = "output.txt"

def GetAddr(ip):
    return f"{ip[0]}.{ip[1]}.{ip[2]}.{ip[3]}:{defaultPort}"

class SearchThread(threading.Thread):
    def __init__(self, ipRange):
        threading.Thread.__init__(self)
        self.ipRange = ipRange
    def run(self):
        asyncio.run(self.async_run())
    async def async_run(self):
        tasks = []
        while self.ipRange[0] != self.ipRange[1]:
            server = JavaServer.lookup(GetAddr(self.ipRange[0]))
            server.timeout = defaultTimeout
            tasks.append([self.ipRange[0].copy(), asyncio.create_task(server.async_status())])

            self.ipRange[0][3] += 1
            for i in reversed(range(len(self.ipRange[0]))):
                if self.ipRange[0][i] > 255:
                    self.ipRange[0][i] = 0
                    self.ipRange[0][i - 1] += 1
                else:
                    break

        for task in tasks:
            try:
                await asyncio.wait_for(task[1], timeout=waitForTimeout)
                task[1].result()
                outputFile = open(outputFileName, 'a')
                outputFile.write(GetAddr(task[0]) + '\n')
                outputFile.close()
            except:
                pass
        print("Thread finished!")

def main():
    print("Program started!")
    for i in range(len(ipIters)):
        print(f"Iteration #{i} started!")
        searchThreads = []
        for ipRange in ipIters[i]:
            searchThreads.append(SearchThread(ipRange))
            searchThreads[len(searchThreads) - 1].start()
        for thread in searchThreads:
            thread.join()

if __name__ == "__main__":
    main()
`

So, this is your code? Let's break it down:

Threading with asyncio: You're mixing threading with asyncio, which is overcomplicating things. There are much cleaner ways to handle concurrency in Python.
Hardcoding IP Ranges: Instead of dynamically generating IPs or using more sophisticated network scanning methods, you're expecting users to input IP ranges manually. That's not only inefficient but also impractical.
Error Handling: Your except block is too broad. Catching all exceptions without any logging or handling is a bad practice and makes debugging a nightmare.
Output Handling: Writing to a file in this manner can lead to file corruption, especially when multiple threads try to write simultaneously. Using a thread-safe method or a logging library would be much better.
In conclusion, if you think this poorly designed script is groundbreaking, you need to reassess your standards. And if your intent was truly educational, you should be more responsible about the potential misuse of your code

and reporting me for the truth is outrageous.

**Comment by FFUV** [comment]
> Seriously, you think you’ve done something big with this repo? 😂 Even one of the main devs of Copenheimer is calling you out. 💀
@RandomGamingDev 

Also wow, I’m truly amazed by your dedication to being the self-appointed enforcer of GitHub etiquette. I didn’t realize we were in the presence of such a distinguished gatekeeper! My sincerest apologies if my previous message didn't meet your high standards of decorum. It’s clear now that your mission to uphold the sanctity of online interactions is as impressive as it is, well, unique.

Rest assured, I’ll make sure to adjust my approach to match your impeccable expectations. If you need any more guidance on how to maintain the delicate balance of GitHub's universe, just let me know.

Wow, I’m truly amazed by your dedication to being the self-appointed enforcer of GitHub etiquette. I didn’t realize we were in the presence of such a distinguished gatekeeper! My sincerest apologies if my previous message didn't meet your high standards of decorum. It’s clear now that your mission to uphold the sanctity of online interactions is as impressive as it is, well, unique.

Rest assured, I’ll make sure to adjust my approach to match your impeccable expectations. If you need any more guidance on how to maintain the delicate balance of GitHub's universe, just let me know. Please just let me know anything you need my Minecraft anarchy king please because reporting me wow you must be the golden balance you must be a god of somewhat. 

But on a serious note reporting me for saying that (which was criticism) is crazy considering you didn't report @orsondmc whilst you were literally glazing him

**Comment by FFUV** [comment]
blue really deleted my comments he can't take my HEAT LMFAO

**Comment by FFUV** [comment]
Still running from the truth deleting my comments.

**Comment by FFUV** [comment]
I have everything screenshot and recorded so don't worry

**Comment by FFUV** [comment]
nah blud is mad!!!
